godtools/App/Share/Domain/UseCases/GetShareableImageUseCase/GetShareableImageUseCase.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  GetShareableImageUseCase.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 7/25/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import Foundation
+import GodToolsToolParser
+
+class GetShareableImageUseCase {
+    
+    func getShareableImage(from shareable: Shareable, manifestResourcesCache: ManifestResourcesCache) -> ShareableImageModel? {
+        guard let shareableImage = shareable as? ShareableImage, let resource = shareableImage.resource, let imageToShare = manifestResourcesCache.getImageFromManifestResources(resource: resource) else {
+            return nil
+        }
+        
+        return ShareableImageModel(
+            image: imageToShare,
+            imageId: shareable.id,
+            toolAbbreviation: shareable.manifest.code
```

Comment:
I noticed that the ```manifest.code``` is an optional value.  Would it be safer to fetch the abbreviation from the resource?  Or fallback to the resource abbreviation if manifest code is null?  @frett thoughts here.

Reply:
on Android I just pass it as null if it doesn't exist. We've been actively publishing the code and the locale in manifests for several years now, it's just older translations that may be missing them.

If you want to fallback to the what is set in your local resource, that's fine

Reply:
https://github.com/CruGlobal/mobile-content-api/pull/159

For reference, we've been including those attributes for almost 4 years now when publishing a translation

Reply:
https://github.com/CruGlobal/mobile-content-api/pull/159

For reference, we've been including those attributes for almost 4 years now when publishing a translation

Reply:
I'm fine with mimicking Android here.